lib/ui/painting.dart

```diff
@@ -12,6 +12,21 @@ part of dart.ui;
 // or invalid value to be used by the engine even in release mode, since that
 // would cause a crash. It is, however, acceptable for error messages to be much
 // less useful or correct in release mode than in debug mode.
+//
+// Painting APIs will also warn about arguments representing NaN or infinite
+// coordinates, which can not be rendered by Skia.
+
+bool rectIsValid(Rect rect) {
```

Comment:
These should probably be private (prefixed with `_` so they're only visible in this file).